File: Scripts/newTask/concurrent-computing/go/concurrent-computing.go
package main

import (
    "fmt"
    "math/rand"
    "time"
)

func main() {
    words := []string{"Enjoy", "Rosetta", "Code"}
    rand.Seed(time.Now().UnixNano())
    q := make(chan string)

    for j := 0; j < 1000000; j++ {

    for _, w := range words {
        go func(w string) {
            time.Sleep(time.Duration(rand.Int63n(1)))
            q <- w
        }(w)
    }
    for i := 0; i < len(words); i++ {
        fmt.Println(<-q)
    }
}
}

File: scripts/task_tested_with_compile_script/concurrent-computing/perl/concurrent-computing.pl
use threads;
use Time::HiRes qw(sleep);

my $r = 1;

sub executeTask {
	my ($i) + @_;

    $_->join for map {
        threads->create(sub {
            #sleep 1;
            print shift,"\n";
        }, $_)
    } qw(Enjoy Rosetta Code);
	return ($i);
}

for( $a=0; $a < 1000000; $a = $a + 1) { 
	$r = executeTask($a);
}
